internal/oci/runtime_oci.go

```diff
@@ -1101,29 +1101,38 @@ func (c *Container) conmonPidFilePath() string {
 // It returns immediately with a channel to find when path had one of those events.
 // done can be used to stop the watch.
 // WatchForFile is responsible for closing all internal channels and the returned channel, but not for closing done.
-func WatchForFile(path string, done chan struct{}, opsToWatch ...notify.Event) (chan struct{}, error) {
-	eiCh := make(chan notify.EventInfo, 1)
+func WatchForFile(path string, done chan struct{}, opsToWatch ...fsnotify.Op) (chan struct{}, error) {
+	watcher, err := fsnotify.NewWatcher()
+	if err != nil {
+		return nil, err
+	}
+
 	ch := make(chan struct{})
 
 	dir := filepath.Dir(path)
-	if err := notify.Watch(dir, eiCh, opsToWatch...); err != nil {
-		return nil, err
-	}
 	go func() {
+		defer watcher.Close()
 		defer close(ch)
-		defer close(eiCh)
-		defer notify.Stop(eiCh)
 		for {
 			select {
```

Comment:
Comparing with the container exit monitor, we are missing a case for wacher.Errors.